pkg/vault/client_test.go

```diff
@@ -0,0 +1,77 @@
+package vault
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/operator-framework/operator-sdk/pkg/log/zap"
+)
+
+func testGetHTTPServer(statusCode int, body []byte) *httptest.Server {
+	testServer := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		res.WriteHeader(statusCode)
+		_, _ = res.Write(body)
+	}))
+
+	return testServer
+}
+
+func TestNewClient(t *testing.T) {
```

Comment:
Maybe add another test for the `SetToken()` create/update path?

Reply:
Added separate test case